angular-workspace/projects/ni/nimble-angular/src/directives/radio-button/nimble-radio-button-control-value-accessor.directive.ts

```diff
@@ -0,0 +1,68 @@
+import { Directive, ElementRef, forwardRef, Host, Inject, Injector, OnInit, Optional, Renderer2 } from '@angular/core';
+// eslint-disable-next-line camelcase
+import { NG_VALUE_ACCESSOR, RadioControlValueAccessor, ɵangular_packages_forms_forms_r } from '@angular/forms';
+import { NimbleRadioGroupDirective } from '../radio-group/nimble-radio-group.directive';
+import type { RadioButton } from './nimble-radio-button.module';
+
+/**
+ * Control Value Accessor implementation for the radio group.
+ */
+@Directive({
+    selector:
+        'nimble-radio-button[formControlName],nimble-radio-button[formControl],nimble-radio-button[ngModel]',
+    // The following host metadata is duplicated from RadioControlValueAccessor
+    // eslint-disable-next-line @angular-eslint/no-host-metadata-property
+    host: { '(change)': 'nimbleOnChange($event.target.checked)', '(blur)': 'onTouched()' },
+    providers: [{
+        provide: NG_VALUE_ACCESSOR,
+        useExisting: forwardRef(() => NimbleRadioButtonControlValueAccessorDirective),
+        multi: true
+    }]
+})
+export class NimbleRadioButtonControlValueAccessorDirective extends RadioControlValueAccessor implements OnInit {
+    private _privateOnChange?: () => void;
+
+    // Type ɵangular_packages_forms_forms_r from base class isn't in camelcase
+    // eslint-disable-next-line camelcase
+    public constructor(renderer: Renderer2, private readonly elementRef: ElementRef, _registry: ɵangular_packages_forms_forms_r, _injector: Injector,
+        @Inject(NimbleRadioGroupDirective) @Optional() @Host() private readonly _radioGroup?: NimbleRadioGroupDirective) {
+        super(renderer, elementRef, _registry, _injector);
+    }
+
+    public ngOnInit(): void {
+        if (this._radioGroup) {
+            (this.elementRef.nativeElement as RadioButton).value = this._radioGroup.allocateId();
```

Comment:
Talked to mert and experimented in a jsbin with setting checked directly on radio items and weren't able to reproduce an issue that needed us to manually update the radio-button-group. Should double check if the pattern is needed. Helper jsbin https://jsbin.com/jidamaxoyo/edit?html,js,output

Reply:
Confirmed that we do need to set the group's value in the CVA's `writeValue`. Added a comment to the code that attempts to explain the situation. I also moved the `allocateId` function from the radio group directive to the CVA, so the CVA no longer needs any reference to the radio group directive.